### DISCUSSION TITLE
[Documentation] What does the Get Document endpoint return?

### DISCUSSION THREAD

**Discussion by nikspyratos** [Investec-Developer-Community/.github#17] [Q&A] [answered]
The response sample just shows an empty data response. I'm assuming this is more likely because the endpoint may return a file download directly. Is this the case? Could the docs add a little more info on what this endpoint will do?

**Comment by stevebissett** [comment, marked as answer]
I noticed this as well - is there any way we can contribute changes to the docs?
But for clarity on the functionality: 

* Retrieve a particular document for the specified date and document type. (PDF only I have found so far)
* Response headers contain `Content-Type:application/pdf`.
* Response headers contain file name in this format:

```
attachment; filename="Statement_04/01/2024 00:00:00.pdf"; filename*=UTF-8''Statement_04%2F01%2F2024%2000%3A00%3A00.pdf
```

You can save files straight to disk/S3 etc. from this. 

Ruby example: 
```
def save_document_to_file(account, document)
  file_name = "#{account.masked_number}_#{document.file_name}"
  directory = "extracted_files/documents/#{account.masked_number}"
  FileUtils.mkdir_p(directory) unless Dir.exist?(directory)

  file_path = "#{directory}/#{file_name}"

  File.open(file_path, "w") do |file|
    file.write(document.file_content)
  end
end
```

**Reply by nikspyratos** [reply]
Indeed. And thanks for the info!

**Comment by devinpearson** [comment]
Suggestions for the docs can be fed back into the team. There has been a few changes and it's probably due for another review
